compiler/rustc_typeck/src/check/method/mod.rs

```diff
@@ -198,6 +200,55 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let pick =
             self.lookup_probe(span, segment.ident, self_ty, call_expr, ProbeScope::TraitsInScope)?;
 
+        if span.edition() < Edition::Edition2021 {
+            if let sym::try_into = segment.ident.name {
+                if let probe::PickKind::TraitPick = pick.kind {
```

Comment:
Given @m-ou-se's comments about cases where the lint does not trigger, I think we should remove this `if` (also below). This will cause us to migrate inherent methods etc. It's a *bit* surprising that this is necessary, but I think that we can still wind up invoking trait methods over inherent methods, if we can do so with fewer autoderefs or other related changes.

Reply:
In the case of json-ext, the lint doesn't appear to fire even when this check is removed. I'm trying to look into why

Reply:
Took a look at ffishim's issue and it seems like it will help resolve that one, at least.

Reply:
What is "ffishim's issue"? :)